### PR TITLE
65131

### PR DESCRIPTION
**Pull request by jakaria-istauk**

### CODE REVIEW

File: includes/Classes/Helper.php [hunk 1]
```diff
@@ -1577,47 +1577,31 @@ public static function eael_get_current_device_by_screen() {
...
-                            ];
-                        }
-                    }
-    
-                endwhile;
-                wp_reset_postdata();
-            }
-        }
+		foreach( $acf_field_groups as $group ){
+			$_acf_fields = acf_get_fields( $group['key'] );
```
Comment (robinwpdeveloper) [inline includes/Classes/Helper.php:1590]: An underscore is sometimes used to indicate that a class property or method is intended to be private or protected

Let's skip `_` at the beginning of the variable name. 
We can name it `$fields`

File: includes/Classes/Helper.php [hunk 2]
```diff
@@ -1577,47 +1577,31 @@ public static function eael_get_current_device_by_screen() {
...
+				foreach( $_acf_fields as $field ) {
+                    // echo_pre( $field );
+					$acf_fields[ $field['name'] ] = [
+						'ID'    => $field['ID'],
+						'key'   => $field['key'],
+						'label' => $field['label'],
+						'name'  => $field['name'],
+						'type'  => $field['type'],
+						'group' => $group['title'],
+					];
```
Comment (robinwpdeveloper) [inline includes/Classes/Helper.php:1601]: Do we get these values all the time?
How about doing a isset / empty check?
E.x.
`
'group' => $group['title'] ?? '',
`

If these (all) values exist all the time then we dont need empty check.

Comment (jakaria-istauk) [reply includes/Classes/Helper.php:1600]: There is low possibility to miss key. but i added checking for title label, name, title

File: includes/Classes/Helper.php [hunk 3]
```diff
@@ -1587,17 +1587,16 @@ public static function get_all_acf_fields() {
 
         $acf_fields = [];
 		foreach( $acf_field_groups as $group ){
-			$_acf_fields = acf_get_fields( $group['key'] );
-			if( !empty( $_acf_fields ) ) {
-				foreach( $_acf_fields as $field ) {
-                    // echo_pre( $field );
+			$default_acf_fields = acf_get_fields( $group['key'] );
+			if( !empty( $default_acf_fields ) ) {
```
Comment (robinwpdeveloper) [inline includes/Classes/Helper.php:1591]: Lets use spacing for if block as well as for `! empty`

**Review by robinwpdeveloper — CHANGES_REQUESTED** [review]

File: includes/Classes/Helper.php [hunk 4]
```diff
@@ -1588,7 +1588,7 @@ public static function get_all_acf_fields() {
         $acf_fields = [];
 		foreach( $acf_field_groups as $group ){
 			$default_acf_fields = acf_get_fields( $group['key'] );
-			if( !empty( $default_acf_fields ) ) {
+			if( ! empty( $default_acf_fields ) ) {
```
Comment (robinwpdeveloper) [inline includes/Classes/Helper.php:1591]: `If () {}`

**Review by robinwpdeveloper — CHANGES_REQUESTED** [review]

File: includes/Classes/Helper.php [hunk 5]
```diff
@@ -1577,18 +1577,18 @@ public static function eael_get_current_device_by_screen() {
 
     public static function get_all_acf_fields() {
 
-        if( ! class_exists( 'ACF' ) || ! function_exists( 'acf_get_field_groups' ) ){
+        if ( ! class_exists( 'ACF' ) || ! function_exists( 'acf_get_field_groups' ) ){
```
Comment (robinwpdeveloper) [inline includes/Classes/Helper.php:1580]:
```
if () {
// code
}
```

**Review by robinwpdeveloper — CHANGES_REQUESTED** [review]

**Review by robinwpdeveloper — APPROVED** [review]

**Pull request merged**